code/modules/instruments/objs/items/headphones.dm

```diff
@@ -19,8 +20,13 @@
 	QDEL_NULL(song)
 	return ..()
 
-/obj/item/clothing/ears/headphones/attack_self(mob/user)
-	ui_interact(user)
+/obj/item/clothing/ears/headphones/ui_action_click(mob/user, actiontype)
+	if(actiontype == /datum/action/item_action/change_headphones_song)
+		ui_interact(user)
+	else
+		on = !on
+		icon_state = item_state = "headphones[on]"
```

Comment:
```suggestion
		icon_state = "headphones[on]"
		item_state = "headphones[on]"
```

Reply:
This is done although the /trigger() action subtype stuff still confuses me sadly.